frontend/src/app/main/ui/ds/controls/select.cljs

```diff
@@ -0,0 +1,245 @@
+             (reset! selected* id)
+             (reset! focused* nil)
+             (reset! open* false)
+             (when (fn? on-change)
+               (on-change id)))))
+
+        options-nodes-refs  (mf/use-ref nil)
+        options-ref         (mf/use-ref nil)
+
+        on-ref
+        (mf/use-fn
+         (fn [node id]
+           (let [refs (or (mf/ref-val options-nodes-refs) #js {})
+                 refs (if node
+                        (obj/set! refs id node)
+                        (obj/unset! refs id))]
+             (mf/set-ref-val! options-nodes-refs refs))))
+
+        on-blur
+        (mf/use-fn
+         (fn [event]
+           (let [click-outside (nil? (.-relatedTarget event))]
+             (when click-outside
+               (reset! focused* nil)
+               (reset! open* false)))))
+
+        on-key-down
+        (mf/use-fn
+         (mf/deps focused disabled)
+         (fn [event]
+           (when-not disabled
+             (let [options (mf/ref-val options-ref)
+                   len     (alength options)
+                   index   (array/find-index #(= (deref focused*) (obj/get % "id")) options)]
+               (dom/stop-propagation event)
+               (cond
+                 (kbd/home? event)
+                 (handle-focus-change options focused* 0 options-nodes-refs)
+
+                 (kbd/up-arrow? event)
+                 (handle-focus-change options focused* (mod (- index 1) len) options-nodes-refs)
+
+                 (kbd/down-arrow? event)
+                 (handle-focus-change options focused* (mod (+ index 1) len) options-nodes-refs)
+
+                 (or (kbd/space? event) (kbd/enter? event))
+                 (when (deref open*)
+                   (dom/prevent-default event)
+                   (handle-selection focused* selected* open*))
+
+                 (kbd/esc? event)
+                 (do (reset! open* false)
+                     (reset! focused* nil)))))))
+
+        class (dm/str class " " (stl/css :select))
+
+        props (mf/spread-props props {:class class
+                                      :role "combobox"
+                                      :aria-controls "listbox"
+                                      :aria-haspopup "listbox"
+                                      :aria-activedescendant focused
+                                      :aria-expanded open
+                                      :on-key-down on-key-down
+                                      :disabled disabled
+                                      :on-click on-click
+                                      :on-blur on-blur})
+
+        selected-option (get-option options selected)
+        label (obj/get selected-option "label")
+        icon (obj/get selected-option "icon")]
+
+    (mf/with-effect [options]
+      (mf/set-ref-val! options-ref options))
+
+    [:div {:class (stl/css :select-wrapper)}
+     [:> :button props
+      [:div {:class (stl/css-case :select-header true
```

Comment:
nitpick: should this be a span? Since `<button>` should have phrasing-content elements as children https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#technical_summary